migrations/V9__Plan_Collaborator_Lead_Trigger.sql

```diff
@@ -0,0 +1,29 @@
+CREATE FUNCTION collaborator_role_check_trigger() RETURNS TRIGGER AS $role_check$
+BEGIN
+ IF (SELECT count(*)
+ FROM plan_collaborator
+ WHERE  team_role = 'MODEL_LEAD' and model_plan_id = OLD.model_plan_id 
+ )<2
+ THEN
+    RAISE EXCEPTION 'There must be at least one MODEL_LEAD assigned to each model plan';
+     END IF;
+     IF (TG_OP = 'DELETE') THEN RETURN OLD;
+     ELSE
+    RETURN NEW;
```

Comment:
Can you explain this `if delete, return old, else return new` block? I don't think I quite understand why we always wouldn't return `OLD`, since the operation always is failed in this `IF` block I think

Reply:
Yeah, so this is looking at the operation that happened. This trigger happens on delete or update statements.  This is basically saying if it was a delete statement, return the row that we just deleted, but if it's an update statement,  return the brand new updated row. Otherwise, we are not returning the new data.